src/videojs-http-streaming.js

```diff
@@ -600,10 +600,46 @@ if (!videojs.use) {
 videojs.m3u8 = m3u8;
 videojs.options.hls = videojs.options.hls || {};
 
+const getTimeRanges = (player, type) => {
+  const timeRangesList = [];
+  const timeRanges = player[type]();
+
+  for (let i = 0; i < timeRanges.length; i++) {
+    timeRangesList.push({
+      start: timeRanges.start(i),
+      end: timeRanges.end(i)
+    });
+  }
+
+  return timeRangesList;
+};
+
+const getPlaybackStats = function() {
+  const player = this;
+
+  return {
+    videoPlaybackQuality: player.getVideoPlaybackQuality(),
+    playerDimensions: player.currentDimensions(),
+    hlsStats: player.tech_.hls.stats,
+    duration: player.duration(),
+    buffered: getTimeRanges(player, 'buffered'),
+    seekable: getTimeRanges(player, 'seekable'),
+    currentTime: player.currentTime(),
+    timestamp: Date.now(),
+    currentSource: player.currentSource(),
+    master: player.tech_.hls.masterPlaylistController_.masterPlaylistLoader_.master,
+    currentTech: player.tech_.name_,
+    // TODO: filter by VHS logging only
```

Comment:
How do we want to go about this? Currently videojs doesn't do any type of filtering or bucketing of the history, just a single array that all logging args get dumped into.

Some options I can think of:
* prefix all log messages we add with a VHS specific string so we can filter on that
* send only objects into the logs e.g. `{ vhs: true, message: 'log message' }` Downside to this is if the debug log is enabled, videojs will console log objects for all of our messages instead of just the message
* extend videojs.log to seperate history by type or allow a way of requesting logs in history of a specific type. Downside to this would be not being available on previous versions of videojs